google/cloud/bigquery/job/query.py

```diff
@@ -1116,6 +1130,16 @@ def dml_stats(self) -> Optional[DmlStats]:
         else:
             return DmlStats.from_api_repr(stats)
 
+    @property
+    def transaction_info(self) -> Optional[TransactionInfo]:
+        """Information of the multi-statement transaction if this job is part of one.
+        """
+        info = self._properties.get("statistics", {}).get("transactionInfo")
+        if info is None:
+            return None
+        else:
+            return TransactionInfo.from_api_repr(info)
```

Comment:
Should this be part of the base job class?

At the time of writing only Query jobs can initiate transactions, but the stats are returned in the top level stats dict, i.e. not under the "query" sub-dict.

Reply:
Answer: Wherever the reservations stats live.